db/query_test.go

```diff
@@ -568,3 +569,128 @@ func countQueryResults(results sgbucket.QueryResultIterator) int {
 	}
 	return count
 }
+
+func TestQueryChannelsActiveOnlyWithLimit(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test require Couchbase Server")
+	}
+
+	db, testBucket := setupTestDB(t)
+	defer testBucket.Close()
+	defer tearDownTestDB(t, db)
+
+	var sequence = 1
+	seqLogEntries := make(map[int]LogEntry)
+	body := Body{"channels": []string{"ABC"}}
+
```

Comment:
If createDocs and deleteDocs are only being used once in this test, they don't need to be an inline function - can just be run as part of setup.  If they are being reused, consider whether they can be defined outside the test (taking t, db as parameters)